src/test/java/spoon/generating/CloneVisitorGenerator.java

```diff
@@ -417,15 +417,30 @@ private boolean avoidThrowUnsupportedOperationException(CtMethod<?> candidate) {
 			 * @param getter <code>getX</code>.
 			 */
 			private CtInvocation<?> createGetterInvocation(CtParameter<?> element, CtMethod<?> getter) {
-				return factory.Code().createInvocation(factory.Code().createVariableRead(element.getReference(), false), getter.getReference());
+				CtInvocation<?> invocation = factory.Code().createInvocation(
+						factory.Code().createVariableRead(element.getReference(), false),
+						getter.getReference()
+				);
+
+				if (getter.getSimpleName().equals("getExtendedModifiers")) {
+					CtExecutableReference<Object> cloneReference = factory.Executable()
+							.createReference(
+									"Set<CtExtendedModifier> #clone(Set<CtExtendedModifier>)"
+							);
+					invocation = factory.Code().createInvocation(
+							null, cloneReference, invocation
+					);
+				}
+
+				return invocation;
```

Comment:
Without the context of the explanation you provided in #4136, this would have me very confused. Even knowing that I was just slightly confused reading this. The big problem for me is that the method doesn't do what it says: it can either create a getter invocation, or create a clone invocation, but it only implies it does the former.

Try to refactor this such that the method name doesn't lie, and make sure to include an explanation for why it's necessary to create a clone call instead of a getter call for extended modifiers. I don't think I would have been able to figure that out.

One option would be to do something like so:

1. Leave `createGetterInvocation` unchanged.
2. Create a new method `createGetterOrCloneInvocation` and call that at the place where `createGetterInvocation` is now called. In this method, either delegate directly to `createGetterInvocation()` or return a clone invocation as appropriate. Make sure to provide an explanation for why the clone invocation is necessary.

That's just one way to do it, you may chose to do something different. As long as the methods do what they say, and there's an explanation for the clone call, it's all good.

Reply:
My motivation to mess with these results of ancient civilizations pursuing forbidding knowledge wasn't especially high and I didn't want the card house to collapse, but I refactored it slightly.

I moved the code out of the getter method as that was complex enough on its own and inlined it in the only callsite with an explanatory comment. I considered extracting it in a method but I found no name I felt actually helped so I left it in the current state. I am very open to suggestions on that front.

Reply:
I fiddled around with a few different variations and landed on something that I think is somewhat clear. See comment.